core/vtk/ttkCompactTriangulationPreconditioning/ttkCompactTriangulationPreconditioning.cpp

```diff
@@ -0,0 +1,203 @@
+  this->SetNumberOfInputPorts(1);
+  this->SetNumberOfOutputPorts(1);
+}
+
+ttkCompactTriangulationPreconditioning::
+  ~ttkCompactTriangulationPreconditioning() {
+}
+
+int ttkCompactTriangulationPreconditioning::FillInputPortInformation(
+  int port, vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkPointSet");
+    return 1;
+  }
+  return 0;
+}
+
+int ttkCompactTriangulationPreconditioning::FillOutputPortInformation(
+  int port, vtkInformation *info) {
+  if(port == 0) {
+    info->Set(ttkAlgorithm::SAME_DATA_TYPE_AS_INPUT_PORT(), 0);
+    return 1;
+  }
+  return 0;
+}
+
+int ttkCompactTriangulationPreconditioning::RequestData(
+  vtkInformation *ttkNotUsed(request),
+  vtkInformationVector **inputVector,
+  vtkInformationVector *outputVector) {
+
+  // Get input object from input vector
+  // Note: has to be a vtkPointSet as required by FillInputPortInformation
+  vtkPointSet *inputDataSet = vtkPointSet::GetData(inputVector[0]);
+  if(!inputDataSet)
+    return 0;
+
+  // If all checks pass then log which array is going to be processed.
+  this->printMsg("Starting computation...");
+
+  ttk::Triangulation *triangulation
+    = ttkAlgorithm::GetTriangulation(inputDataSet);
+  if(!triangulation)
+    return 0;
+
+  // Precondition the triangulation (e.g., enable fetching of vertex neighbors)
+  // this->preconditionTriangulation(triangulation); // implemented in base
+  // class
+
+  // Templatize over the different input array data types and call the base code
+  int status = 0; // this integer checks if the base code returns an error
+  ttkTemplateMacro(
+    triangulation->getType(),
+    (status = this->execute((TTK_TT *)triangulation->getData(), Threshold)));
+
+  // On error cancel filter execution
+  if(status != 1)
+    return 0;
+
+  // Get output vtkDataSet (which was already instantiated based on the
+  // information provided by FillOutputPortInformation)
+  vtkPointSet *outputDataSet = vtkPointSet::GetData(outputVector, 0);
+  vector<SimplexId> vertexMap(this->vertices.size());
+  vtkUnstructuredGrid *outputMesh
+    = vtkUnstructuredGrid::SafeDownCast(outputDataSet);
+
+  vtkSmartPointer<vtkPoints> points = vtkSmartPointer<vtkPoints>::New();
+  vtkSmartPointer<vtkIntArray> indices = vtkSmartPointer<vtkIntArray>::New();
+
+  indices->SetNumberOfComponents(1);
+  indices->SetName(compactTriangulationIndex);
+
+  // insert the vertices in the output mesh
+  for(size_t i = 0; i < this->vertices.size(); i++) {
+    float x, y, z;
+    triangulation->getVertexPoint(this->vertices.at(i), x, y, z);
+    points->InsertNextPoint(x, y, z);
+    vertexMap[this->vertices.at(i)] = i;
+    indices->InsertNextTuple1(this->nodes.at(i));
+  }
+  outputMesh->SetPoints(points);
+
+  vtkPointData *pointData = outputMesh->GetPointData();
+  pointData->AddArray(indices);
+
+  std::vector<std::string>::iterator iter = scalarFields.begin();
+  while(iter != scalarFields.end()) {
```

Comment:
A range-based for loop (`for(const auto &s : scalarFields)`) should be more readable here